org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/BuilderParticipant.java

```diff
@@ -590,9 +587,13 @@ protected void handleChangedContents(Delta delta, IBuildContext context, Eclipse
 		saveResourceStorage(resource, fileSystemAccess);
 		if (shouldGenerate(resource, context)) {
 			try {
+				IProgressMonitor fsaMonitor = fileSystemAccess.getMonitor();
 				MonitorBasedCancelIndicator cancelIndicator = new MonitorBasedCancelIndicator(fileSystemAccess.getMonitor());
```

Comment:
Please use the fsaMonitor for the cancelIndicator, too instead of reading it again from the fileSystemAccess.

Reply:
Done.